packages/yourloops/components/patient-data/patient-data.utils.ts

```diff
@@ -84,6 +86,27 @@ export class PatientDataUtils {
     this.patient = patient
   }
 
+  calculateDashboardDateRange(dateBg: string[]): number {
+    const dateRangeSet = new Set(dateBg)
+    if (dateRangeSet.size >= DEFAULT_DASHBOARD_TIME_RANGE_DAYS) {
+      return DEFAULT_MS_RANGE * DEFAULT_DASHBOARD_TIME_RANGE_DAYS
+    }
+    return dateRangeSet.size * DEFAULT_MS_RANGE
+  }
+
+  getRangeDaysInMs(data: MedicalData): number {
+    if (data.smbg.length !== 0) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏I found a way to simplify the whole method (sorry :joy:):

```ts
const dateRangeData = data.smbg.length > 0 ? data.smbg : data.cbg
const localDates = dateRangeData.map((bgData) => bgData.localDate)

return this.calculateDashboardDateRange(localDates)
```